File: backend-java/src/main/java/com/github/pascualex/oc/repository/CourseRepository.java
package com.github.pascualex.oc.repository;

import com.github.pascualex.oc.mapper.CourseMapper;
import com.github.pascualex.oc.model.Course;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public class CourseRepository {

    private final CourseMapper courseMapper;

    @Autowired
    public CourseRepository(CourseMapper courseMapper) {
        this.courseMapper = courseMapper;
    }

    public List<Course> getCourses() {
        return courseMapper.getCourses();
    }

    public Course getCourse(int id) {
        final List<Course> courses = courseMapper.getCoursesById(id);

        if (courses.size() != 1) return null;

        return courses.get(0);
    }

    public void addCourse(Course course) {
        courseMapper.addCourse(course);
    }

    public Course setCourseActive(int id, boolean active) {
        courseMapper.setCoursesActiveById(id, active);
        return getCourse(id);
    }

    public void removeCourse(int id) {
        courseMapper.removeCoursesById(id);
    }
}

File: backend-node/src/repository/courseRepository.ts
import * as courseMapper from '../mapper/courseMapper';
import { Course, CourseRow, rowToCourse } from '../model/Course';

export async function getCourses(): Promise<Course[]> {
  const rows: CourseRow[] = await courseMapper.getCourses();
  return rows.map((row) => rowToCourse(row));
} 

export async function getCourse(id: number): Promise<Course> {
  const rows: CourseRow[] = await courseMapper.getCoursesById(id);
  if (rows.length > 0) {
    return rowToCourse(rows[0]);
  } else {
    return null;
  }
}

export async function addCourse(course: Course): Promise<void> {
  return courseMapper.addCourse(course);
}

export async function setCourseActive(id: number, active: boolean): Promise<void> {
  return courseMapper.setCoursesActiveById(id, active);
}

export async function removeCourse(id: number): Promise<void> {
  return courseMapper.removeCoursesById(id);
}